boards/sparkfun/micromod/doc/index.rst

```diff
@@ -0,0 +1,202 @@
+| FLASH     | on-chip    | flash                               |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+| USB       | on-chip    | usb                                 |
++-----------+------------+-------------------------------------+
+| QSPI      | on-chip    | nordic,nrf-qspi                     |
++-----------+------------+-------------------------------------+
+
+
+Connections and IOs
+===================
+
+LED
+---
+* Led0 (blue) = P0.13
+
+Micromod header
+---------------
+
+* micromod_1_uart = uart0
+* micromod_2_uart = uart1
+* micromod_0_i2c = i2c0
+* micromod_1_i2c = i2c1
+* micromod_0_spi = spi2
+* micromod_header compatible with ``sparkfun,micromod-gpio``
+
+Programming and Debugging
+*************************
+
+Applications for the ``micromod/nrf52840`` board configuration can be
+built, flashed, and debugged in the usual way. See
+:ref:`build_an_application` and :ref:`application_run` for more details on
+building and running.
+
+Flashing
+========
+
+Follow the instructions in the :ref:`nordic_segger` page to install
+and configure all the necessary software. Further information can be
+found in :ref:`nordic_segger_flashing`. Then build and flash
+applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+The flashing tool will depend on the carrier used along with the board.
+In the case of `Sparkfun asset tracking carrier`, it is possible to use
+the SWD interface along with a J-Link.
+
+Here is an example for the :ref:`hello_world` application.
+
+First, run your favorite terminal program to listen for output.
+
+.. code-block:: console
+
+   $ minicom -D <tty_device> -b 115200
+
+Replace :code:`<tty_device>` with the port where the board nRF52840 DK
+can be found. For example, under Linux, :code:`/dev/ttyACM0`.
+
+Then build and flash the application in the usual way.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: micromod/nrf52840
+   :goals: build flash
+
+Debugging
+=========
+Refer to the :ref:`nordic_segger` page to learn about debugging Nordic boards with a
+Segger IC.
+
+In case of using the jlink for debugging, you may want to use RTT viewer for displaying
+debug messages. In this case, you need to add the following configuration into your
+prj.conf file:
+
+   .. code-block::
+      # Enable RTT
+      CONFIG_CONSOLE=y
+      CONFIG_UART_CONSOLE=n
+      CONFIG_RTT_CONSOLE=y
+      CONFIG_USE_SEGGER_RTT=y
+      CONFIG_SHELL=y
+      CONFIG_SHELL_BACKENDS=y
+      CONFIG_SHELL_BACKEND_RTT=y
+
+Testing the 128Mb qspi memory flash
+************************************
+The ``micromod/nrf52840`` board variant has a built-in NOR flash memory connected
```

Comment:
this is not a variant

Reply:
I change it to SoC since it is just the same board with different SoC